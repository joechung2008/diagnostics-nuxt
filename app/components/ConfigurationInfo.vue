<template>
  <div>
    <h3 class="text-lg font-semibold mb-2">Configuration</h3>
    <UTable :data="data" :columns="columns">
      <template #key-data="{ row }">
        <span class="text-sm">{{ row.key }}</span>
      </template>
      <template #value-data="{ row }">
        <span class="text-sm">{{ row.value }}</span>
      </template>
    </UTable>
  </div>
</template>

<script setup lang="ts">
import type { TableColumn } from "@nuxt/ui";
import type { ConfigurationProps, KeyValuePair } from "~/types";

const props = defineProps<ConfigurationProps>();

const data = computed(() => {
  return Object.entries(props.config).map(([key, value]) => ({
    key,
    value,
  }));
});

const columns: TableColumn<KeyValuePair<string>>[] = [
  {
    id: "key",
    accessorKey: "key",
    header: "Key",
  },
  {
    id: "value",
    accessorKey: "value",
    header: "Value",
  },
];
</script>
