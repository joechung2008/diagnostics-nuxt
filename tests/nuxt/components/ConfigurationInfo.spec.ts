import { mountSuspended, renderSuspended } from "@nuxt/test-utils/runtime";
import { describe, expect, it } from "vitest";
import ConfigurationInfo from "../../../app/components/ConfigurationInfo.vue";
import type { ConfigurationProps } from "../../../app/types";

describe("app/components/ConfigurationInfo.vue", () => {
  it("should mount", async () => {
    const component = await mountSuspended(ConfigurationInfo);
    expect(component).toBeDefined();
  });

  it("should render configuration info", async () => {
    const props: ConfigurationProps = {
      config: {
        database: "postgresql",
        cache: "redis",
        port: "8080",
      },
    };

    const { html } = await renderSuspended(ConfigurationInfo, {
      props,
    });

    expect(html()).toMatchSnapshot();
  });
});
