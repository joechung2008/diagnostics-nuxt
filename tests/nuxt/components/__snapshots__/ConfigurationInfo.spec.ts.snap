// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`app/components/ConfigurationInfo.vue > should render configuration info 1`] = `
"<div id="test-wrapper">
  <div>
    <h3 class="text-lg font-semibold mb-2">Configuration</h3>
    <div class="relative overflow-auto">
      <table class="min-w-full overflow-clip">
        <!--v-if-->
        <thead class="relative">
          <tr class="data-[selected=true]:bg-elevated/50">
            <th data-pinned="false" scope="col" class="px-4 py-3.5 text-sm text-highlighted text-left rtl:text-right font-semibold [&amp;:has([role=checkbox])]:pe-0">Key</th>
            <th data-pinned="false" scope="col" class="px-4 py-3.5 text-sm text-highlighted text-left rtl:text-right font-semibold [&amp;:has([role=checkbox])]:pe-0">Value</th>
          </tr>
          <tr class="absolute z-[1] left-0 w-full h-px bg-(--ui-border-accented)"></tr>
        </thead>
        <tbody class="divide-y divide-default [&amp;>tr]:data-[selectable=true]:hover:bg-elevated/50 [&amp;>tr]:data-[selectable=true]:focus-visible:outline-primary">
          <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
            <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">database</td>
            <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">postgresql</td>
          </tr>
          <!--v-if-->
          <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
            <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">cache</td>
            <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">redis</td>
          </tr>
          <!--v-if-->
          <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
            <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">port</td>
            <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">8080</td>
          </tr>
          <!--v-if-->
        </tbody>
        <!--v-if-->
      </table>
    </div>
  </div>
</div>"
`;
